modules/data.atmosphere/R/met.process.R

```diff
@@ -41,328 +83,98 @@ met.process <- function(site, input_met, start_date, end_date, model, host, dbpa
-                                      in.prefix=new.files$dbfile.name[1],
-                                      siteid = site$id,
-                                      startdate = start_date,
-                                      enddate = end_date,
-                                      mimetype=new.files$mimetype[1],
-                                      formatname=new.files$formatname[1],
-                                      parentid = NA,
-                                      con = con,
-                                      hostname = host$name)
-        if(met %in% "CRUNCEP"){ready.id = raw.id}
-        if(met %in% "GFDL"){ready.id = raw.id}
-      }
-    }else if(register$scale=="site") { # Site-level met
-
-      print("start CHECK")
-      check = db.query(
-        paste0("SELECT i.start_date, i.end_date, d.file_path, d.container_id, d.id  from dbfiles as d join inputs as i on i.id = d.container_id where i.site_id =",site$id,
-               " and d.container_type = 'Input' and i.format_id=",register$format$id, " and d.machine_id =",machine$id,
-               " and (i.start_date <= DATE '",as.POSIXlt(start_date, tz = "GMT"),"') and (DATE '", as.POSIXlt(end_date, tz = "GMT"),"' <= i.end_date)" ),con)
-      print("end CHECK")
-      options(digits=10)
-      print(check)
-      if(length(check)>0){
-        raw.id <- list(input.id=check$container_id, dbfile.id=check$id)
-      }else{
-
-        outfolder = paste0(outfolder,"_site_",str_ns)
-        args <- list(site$name, outfolder, start_date, end_date)
-
-        cmdFcn  = paste0(pkg,"::",fcn,"(",paste0("'",args,"'",collapse=","),paste0(",username='",username,"'"),")")
-        new.files <- remote.execute.R(script=cmdFcn,host=host,user=NA,verbose=TRUE,R="R")
-
-        ## insert database record
-        raw.id <- dbfile.input.insert(in.path=dirname(new.files$file[1]),
-                                      in.prefix=new.files$dbfile.name[1],
-                                      siteid = site$id,
-                                      startdate = start_date,
-                                      enddate = end_date,
-                                      mimetype=new.files$mimetype[1],
-                                      formatname=new.files$formatname[1],
-                                      parentid=NA,
-                                      con = con,
-                                      hostname = host$name)
-      }
+  #--------------------------------------------------------------------------------------------------#
+  # Download raw met
+  if(stage$download.raw) {
+    raw.id <- .download.raw.met.module(
+      dir=dir, met=met, register=register, machine=machine, 
+      start_date=start_date, end_date=end_date, con=con, 
+      input_met=input_met, new.site=new.site, host=host, 
+      overwrite=overwrite$download
+    )
+    if(met %in% c("CRUNCEP", "GFDL")) {
+      ready.id <- raw.id
```

Comment:
OK, doing that now. This is one of those things that met.process should figure out from the registry though, I think.